src/celeritas/track/detail/TrackInitAlgorithms.cu

```diff
@@ -6,46 +6,137 @@
 //---------------------------------------------------------------------------//
 #include "TrackInitAlgorithms.hh"
 
+#if CELERITAS_USE_CUDA
+#    include <cub/device/device_partition.cuh>
+#    include <cub/device/device_scan.cuh>
+#    include <cub/device/device_select.cuh>
+#    include <cub/version.cuh>
+#    if CUB_VERSION >= 200800
+#        include <cub/device/device_transform.cuh>
+#    else
+#        include <thrust/execution_policy.h>
+#        include <thrust/transform.h>
+#    endif
+#elif CELERITAS_USE_HIP
+#    include <hipcub/device/device_partition.hpp>
+#    include <hipcub/device/device_scan.hpp>
+#    include <hipcub/device/device_select.hpp>
+#    include <hipcub/hipcub_version.hpp>
+#    if HIPCUB_VERSION >= 400100
+#        include <hipcub/device/device_transform.hpp>
+#    else
+#        include <thrust/execution_policy.h>
+#        include <thrust/transform.h>
+#    endif
+#endif
 #include <thrust/device_ptr.h>
-#include <thrust/execution_policy.h>
-#include <thrust/partition.h>
-#include <thrust/remove.h>
-#include <thrust/scan.h>
+
+#include "corecel/DeviceRuntimeApi.hh"
 
 #include "corecel/Macros.hh"
+#include "corecel/data/DeviceVector.hh"
 #include "corecel/data/ObserverPtr.device.hh"
-#include "corecel/math/Algorithms.hh"
+// #include "corecel/math/Algorithms.hh" //Use if revert to LogicalNot()
 #include "corecel/sys/Device.hh"
 #include "corecel/sys/ScopedProfiling.hh"
 #include "corecel/sys/Stream.hh"
 #include "corecel/sys/Thrust.device.hh"
 
 #include "../Utils.hh"
 
+#if CELERITAS_USE_CUDA
+using namespace cub;
+#elif CELERITAS_USE_HIP
+using namespace hipcub;
+#endif
+
 namespace celeritas
 {
 namespace detail
 {
 //---------------------------------------------------------------------------//
 /*!
- * Remove all elements in the vacancy vector that were flagged as active
- * tracks.
+ * Create a functor to recognize specific tracks.
  */
+
+//
+//
+template<class T>
+struct NotEqual
+{
+    int compare_;
+
+    NotEqual<T>(int compare) : compare_(compare) {}
```

Comment:
The default constructor should do the same (unless you need to use `NotEqual(1)` instead of `NotEqual{}`
```suggestion
```